test/tests/LayoutGrids.js

```diff
@@ -0,0 +1,593 @@
+        // gridcell div or span
+        if (elementTabindex != null) {
+
+          t.is(
+            elementTabindex,
+            tabindex,
+            '"[role=gridcell]" should have tabindex="-1": ' + ex.gridSelector
+          );
+
+          // Look for a focusable widget
+          let focusableEl = await gridcellElements[el].findElement(focusableWidget[0]).catch(() => {}) ||
+            await gridcellElements[el].findElement(focusableWidget[1]).catch(() => {});
+
+          t.falsy(
+            focusableEl,
+            '"[role=gridcell]" with tabindex="-1" should not contain focusable widgets: ' + ex.gridSelector
+          );
+        }
+
+        // If attribute "tabindex" has not been set, there must be a focusable widgets
+        // within the gridcell div or span with tabindex appropriately set
+        else {
+
+        // Find the focusable widget
+          let widgetElement = await gridcellElements[el].findElement(focusableWidget[0]).catch(() => {}) ||
+          await gridcellElements[el].findElement(focusableWidget[1]).catch(() => {});
+
+          t.truthy(
+            widgetElement,
+            '["role=gridcell]" without tabindex set must contain a focusable widget: ' + ex.gridSelector
+          );
+
+          t.is(
+            await widgetElement.getAttribute('tabindex'),
+            tabindex,
+            'Widget within element "[role=gridcell]" must have tabindex set in example: ' + ex.gridSelector
+          );
+        }
+      }
+    }
+  });
+
+
+// Keys
+
+ariaTest('Right arrow key moves focus', 'grid/LayoutGrids.html', 'key-right-arrow', async (t) => {
+  t.plan(67);
+
+  const cellSelectors = {
+    ex1: '#ex1 [role="gridcell"]',
+    ex2: '#ex2 [role="row"] [role="gridcell"]',
+    ex3: '#ex3 [role="row"] [role="gridcell"]'
+  };
+
+  for (let [exId, selector] of Object.entries(cellSelectors)) {
+    let gridcellElements = await t.context.session.findElements(
+      t.context.By.css(selector)
+    );
+
+    // Find the first focusable element
+    let activeElement = await t.context.session.executeScript(focusWithin, gridcellElements[0]);
+    if (!activeElement) {
+      throw new Error('Could not focus on element or any decendent in the first gridcell: ' + selector);
+    }
+
+    // Test focus moves to next element on arrow key
+
+    for (let index = 1; index < gridcellElements.length; index++) {
+
+      await activeElement.sendKeys(t.context.Key.ARROW_RIGHT);
+      let correctActiveElement = await t.context.session.executeScript(checkActiveElement, selector, index);
+
+      t.truthy(
+        correctActiveElement,
+        'Example ' + exId + ': gridcell `' + (await gridcellElements[index].getText()) + '` should receive focus after arrow right'
+      );
+
+      activeElement = await t.context.session.executeScript(() => {
+        return document.activeElement;
+      });
+    }
+
+    // Test arrow key on final element
```

Comment:
Here you can use the data-wrap-cols attribute to check whether or not the arrows should wrap